src/server/engine_shard_set.cc

```diff
@@ -56,6 +71,67 @@ EngineShard::Stats& EngineShard::Stats::operator+=(const EngineShard::Stats& o)
   return *this;
 }
 
+// This function checks 3 things:
+// 1. Don't try memory fragmentation if we don't use "enough" memory (control by
+// mem_defrag_threshold flag)
+// 2. That we had change in memory usage - to prevent endless loop - out of scope for now
+// 3. in case the above is OK, make sure that we have a "gap" between usage and commited memory
+// (control by commit_use_threshold flag)
+bool EngineShard::DefragTaskArgs::IsRequired() {
+  static const uint64_t mem_size = max_memory_limit;
+  static const float threshold_mem = mem_size / GetFlag(FLAGS_mem_defrag_threshold);
+  static const float commit_use_threshold = GetFlag(FLAGS_commit_use_threshold);
+
+  if (cursor > 0) {
+    return true;
+  }
+
+  uint64_t commited = GetMallocCurrentCommitted();
+  uint64_t mem_in_use = used_mem_current.load(memory_order_relaxed);
+
+  // we want to make sure that we are not running this to many times - i.e.
+  // if there was no change to the memory, don't run this
+  if (threshold_mem < commited && mem_in_use != 0 &&
+      (float(mem_in_use * commit_use_threshold) < commited)) {
```

Comment:
i think that numerically speaking, it's better to change `commit_use_threshold` to double and then here use
`uint64_t(mem_in_use * commit_use_threshold) < commited`, i.e. to compare uint64 vs uint64.

Reply:
OK